### PR TITLE
Feat: PHPstan 2

### PR DESCRIPTION
**Pull request by Rom1-B**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

File: src/NotificationTargetCommonITILObject.php [hunk 1]
```diff
@@ -1090,13 +1090,13 @@ public function addDataForTemplate($event, $options = [])
...
-     * @param CommonDBTM $item    Object instance
-     * @param array      $options Options
-     * @param boolean    $simple  (false by default)
+     * @param CommonITILObject  $item    Object instance
+     * @param array             $options Options
+     * @param boolean           $simple  (false by default)
      *
      * @return array
      **/
-    public function getDataForObject(CommonDBTM $item, array $options, $simple = false)
```
Comment (trasher) [inline src/NotificationTargetCommonITILObject.php:1099]: While this is correct, I do not know if such a change should not be considered as a BC break :/

File: src/NetworkAlias.php [hunk 2]
```diff
@@ -389,10 +389,10 @@ public static function showForNetworkName(NetworkName $item, $withtemplate = 0)
     /**
      * Show the aliases contained by the alias
      *
-     * @param CommonGLPI $item          the FQDN owning the aliases
+     * @param FQDN       $item          the FQDN owning the aliases
      * @param integer    $withtemplate  withtemplate param
      **/
-    public static function showForFQDN(CommonGLPI $item, $withtemplate)
+    public static function showForFQDN(FQDN $item, $withtemplate)
```
Comment (trasher) [inline src/NetworkAlias.php:395]: While this is correct, I do not know if such a change should not be considered as a BC break :/

Comment (Rom1-B) [reply src/NetworkAlias.php:395]: It fixes these errors, so can we use CommonDBTM instead? Which is probably less risky, but seems less relevant to me.
_The function is only called once in the core_
```
src/NetworkAlias.php:402:Call to an undefined method CommonGLPI::check().
src/NetworkAlias.php:402:Call to an undefined method CommonGLPI::getID().
src/NetworkAlias.php:415:Call to an undefined method CommonGLPI::getID().
src/NetworkAlias.php:441:Access to an undefined property CommonGLPI::$fields.
src/NetworkAlias.php:461:Call to an undefined method CommonGLPI::getID().
```

File: src/NotificationTargetCommonITILObject.php (hunk 1 above)
Comment (Rom1-B) [reply src/NotificationTargetCommonITILObject.php:1099]: It fixes these errors:
```
src/NotificationTargetCommonITILObject.php:1148:Call to an undefined method CommonDBTM::getStatus().
src/NotificationTargetCommonITILObject.php:1150:Call to an undefined method CommonDBTM::getUrgencyName().
src/NotificationTargetCommonITILObject.php:1151:Call to an undefined method CommonDBTM::getImpactName().
src/NotificationTargetCommonITILObject.php:1152:Call to an undefined method CommonDBTM::getPriorityName().
src/NotificationTargetCommonITILObject.php:1153:Call to an undefined method CommonDBTM::getActionTime().
src/NotificationTargetCommonITILObject.php:1172:Call to an undefined method CommonDBTM::countUsers().
src/NotificationTargetCommonITILObject.php:1174:Call to an undefined method CommonDBTM::getUsers().
src/NotificationTargetCommonITILObject.php:1203:Call to an undefined method CommonDBTM::countSuppliers().
src/NotificationTargetCommonITILObject.php:1205:Call to an undefined method CommonDBTM::getSuppliers().
src/NotificationTargetCommonITILObject.php:1238:Call to an undefined method CommonDBTM::countUsers().
src/NotificationTargetCommonITILObject.php:1240:Call to an undefined method CommonDBTM::getUsers().
src/NotificationTargetCommonITILObject.php:1261:Call to an undefined method CommonDBTM::countGroups().
src/NotificationTargetCommonITILObject.php:1263:Call to an undefined method CommonDBTM::getGroups().
src/NotificationTargetCommonITILObject.php:1276:Call to an undefined method CommonDBTM::countGroups().
src/NotificationTargetCommonITILObject.php:1278:Call to an undefined method CommonDBTM::getGroups().
src/NotificationTargetCommonITILObject.php:1291:Call to an undefined method CommonDBTM::countUsers().
src/NotificationTargetCommonITILObject.php:1293:Call to an undefined method CommonDBTM::getUsers().
src/NotificationTargetCommonITILObject.php:1311:Call to an undefined method CommonDBTM::countGroups().
src/NotificationTargetCommonITILObject.php:1313:Call to an undefined method CommonDBTM::getGroups().
src/NotificationTargetCommonITILObject.php:1422:Call to an undefined method CommonDBTM::getSolvedStatusArray().
src/NotificationTargetCommonITILObject.php:1423:Call to an undefined method CommonDBTM::getClosedStatusArray().
src/NotificationTargetCommonITILObject.php:1448:Call to an undefined method CommonDBTM::getAssociatedDocumentsCriteria().
src/NotificationTargetCommonITILObject.php:1524:Call to an undefined method CommonDBTM::getActionTime().
src/NotificationTargetCommonITILObject.php:1616:Call to an undefined method CommonDBTM::getTimelineItems().
```

File: src/CommonDBChild.php [hunk 3]
```diff
@@ -248,7 +248,7 @@ public function canChildItem($methodItem, $methodNotItem)
      * @param $getFromDB   (true by default)
      * @param $getEmpty    (true by default)
      *
-     * @return object of the concerned item or false on error
+     * @return bool|CommonDBTM of the concerned item or false on error
```
Comment (AdrienClairembault) [inline src/CommonDBChild.php:251]:
```suggestion
     * @return CommonDBTM|false of the concerned item or false on error
```

File: src/NetworkAlias.php (hunk 2 above)
Comment (AdrienClairembault) [reply src/NetworkAlias.php:395]: I guess (for once) @var FQDN would be enough here.
FYI on main the type has already been fixed.

Comment (trasher) [reply src/NetworkAlias.php:395]: Well.. Problem is potential call from plugins; not from core. But calling with something else than a `FQDN` object will fail so... I guess we can change finally.
Let's see for other reviewers pov ;)

**Review by trasher — APPROVED** [review]

File: src/NotificationTargetCommonITILObject.php (hunk 1 above)
Comment (AdrienClairembault) [reply src/NotificationTargetCommonITILObject.php:1099]: Agreed with @trasher

Comment (orthagh) [reply src/NotificationTargetCommonITILObject.php:1099]: Hum we didn't discuss that point last Thursday.
Is the 10.0bf target mandatory?
for next pr, as complexity increase, maybe we should target main branch, no?

Comment (trasher) [reply src/NotificationTargetCommonITILObject.php:1099]: The method signature chzange does not seems required; changing only phpdoc is enough. This is not "complex"; method signature must be preserved; code must change as less as possible.

Comment (trasher) [reply src/NotificationTargetCommonITILObject.php:1099]: Targeting 10.0 is **not** mandatory at all. But working on phpstan issues already fixed few issues.

Comment (trasher) [reply src/NotificationTargetCommonITILObject.php:1099]: (and oh, I forget to mention; this is not mandatory we reach phpstan level 2 on bugfixes branch)

Comment (Rom1-B) [reply src/NotificationTargetCommonITILObject.php:1099]: In this case, should I revert the file to its original state? That’s probably the simplest solution.
_For your information, there are still around 200 level 2 errors, many of which are similar._

Comment (trasher) [reply src/NotificationTargetCommonITILObject.php:1099]: Does phpdoc change cause an issue? If not, you can keep it.
Method signature change must be reverted, yes.

Comment (Rom1-B) [reply src/NotificationTargetCommonITILObject.php]: Yes, I've just committed it.

File: src/NetworkAlias.php (hunk 2 above)
Comment (trasher) [reply src/NetworkAlias.php:395]: Same here, method signature should stay as it was. Just changing phpdoc should be enough as well at this point.

**Review by trasher — APPROVED** [review]

**Pull request merged**